share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -30,32 +30,36 @@ macro(rocm_create_package)
         set( CPACK_SET_DESTDIR ON CACHE BOOL "Boolean toggle to make CPack use DESTDIR mechanism when packaging" )
     endif()
 
-    if (EXISTS "/etc/os-release")
-        rocm_set_os_id(_os_id)
-        rocm_read_os_release(_version_id "VERSION_ID")
-
-        #only set CPACK_SYSTEM_NAME for AMD supported OSes
-        if (_os_id_centos OR _os_is_rhel)
-            STRING(CONCAT _SYSTEM_NAME "el" ${_version_id} ".x86_64")
-        #Debs use underscrore between OS and architecture
-        elseif(_os_id_ubuntu)
-            STRING(CONCAT _SYSTEM_NAME ${_os_id} "-" ${_version_id} "_amd64")
-        else()
-        #For SLES and unsupported OSes
-            STRING(CONCAT _SYSTEM_NAME ${_os_id} "-" ${_version_id} ".amd64")
-        endif()
-
-        set(CPACK_SYSTEM_NAME ${_SYSTEM_NAME} CACHE STRING "CPACK_SYSTEM_NAME for packaging")
-    endif()
-
     set(CPACK_DEBIAN_PACKAGE_MAINTAINER ${PARSE_MAINTAINER})
     set(CPACK_DEBIAN_PACKAGE_SECTION "devel")
+    set(CPACK_DEBIAN_FILE_NAME "DEB-DEFAULT")
 
     set(CPACK_NSIS_MODIFY_PATH On)
     set(CPACK_NSIS_PACKAGE_NAME ${PARSE_NAME})
 
     set(CPACK_RPM_PACKAGE_RELOCATABLE Off)
-    set( CPACK_RPM_PACKAGE_AUTOREQPROV Off CACHE BOOL "turns off rpm autoreqprov field; packages explicity list dependencies" )
+    set(CPACK_RPM_PACKAGE_AUTOREQPROV Off CACHE BOOL "turns off rpm autoreqprov field; packages explicity list dependencies")
+    set(CPACK_RPM_FILE_NAME "RPM-DEFAULT")
+
+    rocm_get_debian_version_info(DEBIAN_VERSION)
+    if(DEBIAN_VERSION STREQUAL "unknown")
+        set(DEBIAN_VERSION  ${PROJECT_VERSION_TWEAK})
+    endif()
+    rocm_get_rpm_release_info(RPM_RELEASE)
+    if(RPM_RELEASE STREQUAL "unknown")
+        set(RPM_RELEASE  ${PROJECT_VERSION_TWEAK})
+    endif()
+
+    # '%{?dist}' breaks manual builds on debian systems due to empty Provides
+    execute_process(COMMAND rpm --eval %{?dist}
+                    RESULT_VARIABLE PROC_RESULT
+                    OUTPUT_VARIABLE EVAL_RESULT
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if (PROC_RESULT EQUAL "0" AND NOT EVAL_RESULT STREQUAL "")
+           string (APPEND RPM_RELEASE "%{?dist}")
+    endif()
+    set(CPACK_DEBIAN_PACKAGE_RELEASE ${DEBIAN_VERSION})
+    set(CPACK_RPM_PACKAGE_RELEASE ${RPM_RELEASE})
```

Comment:
These should be cached variables so they can be overridden without needing to use env variables

Reply:
Updated as discussed.